456372: assert FD_ISSET unnecessary in function slapd_clr_write

Source: upstream, http://www.openldap.org/devel/cvsweb.cgi/servers/slapd/daemon.c.diff?r1=1.418&r2=1.419&hideattic=1&sortbydate=0
Reviewed-By: Jan Safranek <jsafrane@redhat.com>

diff -up openldap-2.2.13/servers/slapd/daemon.c.old openldap-2.2.13/servers/slapd/daemon.c
--- openldap-2.2.13/servers/slapd/daemon.c.old	2004-04-12 20:13:21.000000000 +0200
+++ openldap-2.2.13/servers/slapd/daemon.c	2008-08-29 15:22:38.000000000 +0200
@@ -305,8 +305,10 @@ void slapd_remove(ber_socket_t s, int wa
 void slapd_clr_write(ber_socket_t s, int wake) {
 	ldap_pvt_thread_mutex_lock( &slap_daemon.sd_mutex );
 
-	assert( FD_ISSET( s, &slap_daemon.sd_actives) );
-	FD_CLR( s, &slap_daemon.sd_writers );
+	if ( FD_ISSET( s, &slap_daemon.sd_writers ) ) {
+		assert( FD_ISSET( s, &slap_daemon.sd_actives) );
+		FD_CLR( s, &slap_daemon.sd_writers );
+	}
 
 	ldap_pvt_thread_mutex_unlock( &slap_daemon.sd_mutex );
 	WAKE_LISTENER(wake);
